src/lets_be_rational/so_rational.rs

```diff
@@ -140,7 +140,7 @@ fn normalised_implied_volatility_from_a_transformed_rational_guess_with_limited_
     // Map in-the-money to out-of-the-money
     let mut beta = beta;
     let mut q = option_type;
-    if q as i32 as f64 * x > 0.0 {
+    if f64::from(q) * x > 0.0 {
```

Comment:
:)

Reply:
whatever it is into or from, just should be consistent

Reply:
Yeah the thing is converting that to "into" would have required either a temporary variable or `Into::<f64>::into(q)`, both of which I thought was uglier than just keeping the from. In any case I just decided to go in a different direction altogether and implement `OptionType * f64`

Reply:
### OptionType Multiplication Performance Comparison

| Input Size | mul_impl (ns) | cast_mul (ns) | Improvement |
|------------|---------------|---------------|-------------|
| 10         | 5.3338        | 6.7369        | 20.82%      |
| 100        | 52.200        | 67.349        | 22.49%      |
| 1000       | 530.66        | 682.93        | 22.29%      |

### results:

The implementation of multiplication is 20% faster than casting. Therefore, comparing (cmp) is faster than transforming i8 to f64.

Reply:
@claireratigan I agree with you, now it looks pure :)